go/interpreter/lfvm/instructions.go

```diff
@@ -1055,45 +1058,48 @@ func genericCall(c *context, kind tosca.CallKind) error {
 		return err
 	}
 
-	baseGas := tosca.Gas(0)
 	// from berlin onwards access cost changes depending on warm/cold access.
 	if c.isAtLeast(tosca.R09_Berlin) {
-		baseGas += getAccessCost(c.context.AccessAccount(toAddr))
-	}
-	checkGas := func(cost tosca.Gas) bool {
-		return 0 <= cost && cost <= c.gas
-	}
-	if !checkGas(baseGas) {
-		return errOutOfGas
+		if err := c.useGas(getAccessCost(c.context.AccessAccount(toAddr))); err != nil {
+			return err
+		}
 	}
 
 	// for static and delegate calls, the following value checks will always be zero.
 	// Charge for transferring value to a new address
 	if !value.IsZero() {
-		baseGas += CallValueTransferGas
-	}
-	if !checkGas(baseGas) {
-		return errOutOfGas
+		if err := c.useGas(CallValueTransferGas); err != nil {
+			return err
+		}
 	}
 
 	// EIP158 states that non-zero value calls that create a new account should
 	// be charged an additional gas fee.
 	if kind == tosca.Call && !value.IsZero() && !c.context.AccountExists(toAddr) {
-		baseGas += CallNewAccountGas
-	}
-	if !checkGas(baseGas) {
-		return errOutOfGas
+		if err := c.useGas(CallNewAccountGas); err != nil {
+			return err
+		}
 	}
 
-	cost := callGas(c.gas, baseGas, provided_gas)
-	if err := c.useGas(baseGas + cost); err != nil {
+	// The Homestead hard-fork introduced a limit on the amount of gas that can be
+	// forwarded to recursive calls. EIP-150 (https://eips.ethereum.org/EIPS/eip-150)
+	// defines that at but one 64th of the available gas in one scope may be passed
+	// to a nested call.
+	endowment := tosca.Gas(c.gas - c.gas/64)
+	if provided_gas.IsUint64() && (endowment >= tosca.Gas(provided_gas.Uint64())) {
+		endowment = tosca.Gas(provided_gas.Uint64())
+	}
+	if err := c.useGas(endowment); err != nil {
+		// this usage can never fail because, since the endowment is at most
+		// 63/64 of the current gas level.
 		return err
 	}
 
+	nestedCallGas := endowment
```

Comment:
Should this renaming here be removed and let it be called either `nestedCallGas` or `endowment` throughout the function?